analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java

```diff
@@ -644,4 +656,124 @@ private void writeRepositorySettingsFile(Map<ArtifactRepository, Path> repositor
+                final String originalModuleGav = ga + ":" + (correspondingModule.getOriginalVersion() == null
+                        ? ""
+                        : correspondingModule.getOriginalVersion());
+                final String newModuleGav = ga + ":" + correspondingModule.getVersion();
+                append(builder, "\tDependencies : {} --> {}", originalModuleGav, newModuleGav);
+                final ModulesItem module = new ModulesItem();
+                module.getGav().setOriginalGAV(originalModuleGav);
+                module.getGav().setPVR(SimpleProjectVersionRef.parse(newModuleGav));
+                modules.add(module);
+                final Map<String, ProjectVersionRef> dependencies = module.getDependencies();
+                final boolean reportNonAligned = configuration.reportNonAligned();
+                allModuleDependencies.forEach((d, p) -> {
+                    final ProjectVersionRef newDependencyVersion = correspondingModule.getAlignedDependencies()
+                            .get(d.toString());
+                    logger.debug("In module {} with GAV {} found a replacement version of {}",
+                            correspondingModule.getProjectPathName(), p, newDependencyVersion);
+
+                    if (newDependencyVersion == null) {
+                        if (reportNonAligned) {
+                            nonAligned.add(d);
+                        }
+                    } else {
+                        final String originalDependencyGav = String.format("%s:%s:%s", p.getGroupId(), p.getArtifactId(),
+                                p.getVersionString());
+                        final String newDependencyGav = String.format("%s:%s:%s", newDependencyVersion.getGroupId(),
+                                newDependencyVersion.getArtifactId(), newDependencyVersion.getVersionString());
+                        append(builder, "\tDependencies : {} --> {}", originalDependencyGav, newDependencyGav);
+                        dependencies.put(d.toString(), newDependencyVersion);
+                    }
+                });
+            }
+
+            builder.append(LINE_SEPARATOR);
+        }
+
+        if (!nonAligned.isEmpty()) {
+            final String type = "Dependencies";
+
+            for (ProjectVersionRef na : nonAligned) {
+                logger.debug("Non-Aligned {} : {}", type, na);
+                append(builder, "\tNon-Aligned {} : {}", type, na);
+            }
+
+            builder.append(LINE_SEPARATOR);
+        }
+
+        final String reportText = builder.toString();
+        logger.info("{}{}", System.lineSeparator(), reportText);
+
+        if (!StringUtils.isEmpty(configuration.reportTxtOutputFile())) {
+            writeReport(rootDir.toPath(), configuration.reportTxtOutputFile(), reportText);
+        }
+
+        if (!StringUtils.isEmpty(configuration.reportJsonOutputFile())) {
+            writeReport(rootDir.toPath(), configuration.reportJsonOutputFile(),
```

Comment:
Shouldn't this output to the build directory not just `rootDir` ? (i.e. as per https://github.com/release-engineering/pom-manipulation-ext/blob/master/core/src/main/java/org/commonjava/maven/ext/core/ManipulationManager.java#L232 )

Reply:
I saw GME writing all other files to `rootDir` I guess that's where I got that, but see the above comment about `@DefaultValue`.

It looks like in PME this value is _always_ relative, in which case I should always resolve the property relative to the output directory. Fixing the location of the output is a separate issue.

Reply:
Other files writing to the root directory need to be committed by repour - thats the difference.

Reply:
Since the value is non-empty by default, `StringUtils.isEmpty(configuration.reportJsonOutputFile())` would be `true` iff the user has disabled it (via a system property). These is a test that checks that this is possible.